### DISCUSSION TITLE
arduino code Self-Balancing-Cube

### DISCUSSION THREAD

**Discussion by nonorobolive** [community/community#77439] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hello everyone,
I'm looking closely at this excellent project of the cube which stands alone in balance. Being new to Arduino programming, I don't understand how to enter the codes provided in the project. There is an arduino_cube.h code, an arduino_cube.ino and a function.ino.
I don't know how to enter the code into an Arduino Nano?
Can someone help me.
Thanks in advance

**Comment by kdinthehouse** [comment]
Download and install the Arduino IDE from https://www.arduino.cc/en/software and choose the appropriate version for your operating system.
Connect your Arduino Nano to your computer
Locate the downloaded "arduino_cube.h", "arduino_cube.ino", and "function.ino" files.

Uploading the code:

Open Arduino IDE: Launch the Arduino IDE application.
Open the main code file: Click on File - Open and select the arduino_cube.ino file.
Select the board and port:
Tools - Board - Arduino Nano
Tools - Port - COM (the port number corresponding to your Nano)
Add the header and function files:
Click Sketch - Include Library - Add .ZIP Library select both arduino_cube.h and function. Ino files and open them and upload the code
Wait for upload completion and verify the upload file by Opening the serial monitor

**Reply by nonorobolive** [reply]
Good morning,
Thanks for your answer, but I don't understand which ZIP library I should add!
Where do I find it? or how do I create it?

**Reply by kdinthehouse** [reply]
compress arduino_cube.h and function.ino to a zip file and upload it

**Reply by nonorobolive** [reply]
Good morning,
I tried, I copied the two files, I compress them in a ZIP file which I rename arduino_cube.h because the program asks for this library but when I want to put the library it gives me this error message:
Error: 13 INTERNAL: Library install failed: archive is not valid: multiple files found in zip file top level
Thanks for your help

**Reply by kdinthehouse** [reply]
don't change .zip keep it as it is

**Reply by nonorobolive** [reply]
This is what I do,
I copy the two files given in the project which are: function.ino and arduino_cube.h
I ZIP them which gives me the arduino_cube Zipper file
I put this file in the library directory but when I want to include the library with the IDE it gives me this message:
Error: 13 INTERNAL: Library install failed: archive is not valid: multiple files found in zip file top level
I don't know much, but I saw that a library necessarily has an h file (this one is there) and a cpp file (this one is not there)

**Reply by nonorobolive** [reply]
Hello everyone,
I will come back to you on this library problem.
I succeeded by putting the arduino_cube.h file and the functions.ino file in tabs next to the main program in the Arduino IDE.
By doing this, compiling and uploading works fine.
I will now order the rest of the parts and 3D print them to continue this project.
Thanks to everyone who helped me.

**Reply by nonorobolive** [reply]
Good morning,
I have just assembled the cube with the Arduino NANO card, but I cannot complete the initial procedure. The video explains with the ESP32 program but I can't understand for the NANO.

**Comment by warundev** [comment]
general steps:

Install Arduino IDE:

If you haven't already, download and install the Arduino IDE from the official Arduino website: [Arduino Software](https://www.arduino.cc/en/Main/Software).
Connect Arduino Nano:

Connect your Arduino Nano to your computer using a USB cable.
Open Arduino IDE:

Open the Arduino IDE on your computer.
Configure Arduino IDE for Arduino Nano:

Go to the "Tools" menu.
Select "Board" and choose "Arduino Nano" or the specific Nano variant you have.
Choose the correct port under the "Port" menu. It will typically be something like "COMx" on Windows or "/dev/cu.usbmodemxxxx" on macOS.
Open the Cube Project Files:

Open the files for your cube project in the Arduino IDE. Typically, the main file is the one with the .ino extension.
Check Dependencies:

Sometimes, Arduino projects have multiple files or dependencies. Make sure that all the necessary files are open in the Arduino IDE.
Upload the Code:

Click on the right arrow icon (Upload) in the Arduino IDE toolbar. This will compile the code and upload it to your Arduino Nano.
Wait for Upload:

The Arduino IDE will show the progress of the upload in the bottom status bar. Wait for the process to complete.
Verify Operation:

Once the upload is successful, the Arduino Nano should start running the code. You may need to power your cube according to the project specifications.
Troubleshooting:

If there are any errors during the upload, read the error messages in the Arduino IDE. They can provide clues about what might be wrong. Common issues include selecting the wrong board or port.
Remember, the main .ino file is usually the starting point, and it includes other files or libraries as needed. Ensure that all necessary files are open in the Arduino IDE before uploading.

**Comment by Limitinit** [comment]
Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
